UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentFileShareServiceTest.cs

```diff
@@ -132,23 +137,56 @@ public async Task WhenRequestDownloadReadMeFile_ThenReturnsTrueIfFileIsDownloade
         {
             bool isFileDownloaded = false;
-            string batchId = "7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272";
-            string exchangeSetRootPath = @"D:\\Downloads";
+            string batchId = "7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272";            
             string filePath = "TestFilePath";
-
             A.CallTo(() => fakefileShareService.DownloadReadMeFile(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(isFileDownloaded);
-            isFileDownloaded = await fulfilmentFileShareService.DownloadReadMeFile(filePath, batchId, exchangeSetRootPath,null);
-
+            isFileDownloaded = await fulfilmentFileShareService.DownloadReadMeFile(filePath, batchId, fakeExchangeSetRootPath, null);
             Assert.AreEqual(false, isFileDownloaded);
         }
+
+        [Test]
+        public async Task WhenValidUploadZipFileRequest_ThenReturnTrue()
+        {
+            fakeIsFileUploaded = true;                   
+            A.CallTo(() => fakefileShareService.UploadZipFileForExchangeSetToFileShareService(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(fakeIsFileUploaded);
+            fakeIsFileUploaded = await fulfilmentFileShareService.UploadZipFileForExchangeSetToFileShareService(fakeBatchId, fakeExchangeSetRootPath, null);
+            Assert.AreEqual(true, fakeIsFileUploaded);
+        }
+
+        [Test]
+        public async Task WhenInvalidUploadZipFileRequest_ThenReturnFalse()
+        {
+            fakeIsFileUploaded = false;                       
+            A.CallTo(() => fakefileShareService.UploadZipFileForExchangeSetToFileShareService(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(fakeIsFileUploaded);
+            fakeIsFileUploaded = await fulfilmentFileShareService.UploadZipFileForExchangeSetToFileShareService(fakeBatchId, string.Empty, null);
+            Assert.AreEqual(false, fakeIsFileUploaded);
+        }
+
+        [Test]
+        public void WhenValidCreateZipFileRequest_ThenReturnTrue()
+        {
+            fakeIsZipFileCreated = true;           
+            A.CallTo(() => fakefileShareService.CreateZipFileForExchangeSet(A<string>.Ignored, A<string>.Ignored)).Returns(fakeIsZipFileCreated);
+            fakeIsZipFileCreated = fulfilmentFileShareService.CreateZipFileForExchangeSet(fakeExchangeSetRootPath, null);
+            Assert.AreEqual(true, fakeIsZipFileCreated);
+        }
+
+        [Test]
+        public void WhenInvalidCreateZipFileRequest_ThenReturnFalse()
+        {
+            fakeIsZipFileCreated = false;            
+            A.CallTo(() => fakefileShareService.CreateZipFileForExchangeSet(A<string>.Ignored, A<string>.Ignored)).Returns(fakeIsZipFileCreated);
+            fakeIsZipFileCreated = fulfilmentFileShareService.CreateZipFileForExchangeSet(string.Empty, null);
+            Assert.AreEqual(false, fakeIsZipFileCreated);
+        }
```

Comment:
As above should this be a single test with an appropriate name?

e.g. CreateZipFileForExchangeSet_CallsFileShareServiceCreateZipFileForExchangeSet

Reply:
In one test case, we are passing fake path (valid request) for parameter ExchangeSetRootPath, where test case will pass and return true. 
and in other test case we are passing empty string (Invalid request) for parameter ExchangeSetRootPath, where test case will fail and return false. Hence we have written two test cases.